### PR TITLE
Fix #1893 Add support for Editing Host Team name

### LINKED ISSUE #1893: Add support for Editing Host Team name
### Observed Behaviour
Currently , there is no option to edit Host Team name.

### Expected Behaviour
Add support for editing challenge host team name.

### Would you like to work on this?
Yes.

@deshraj @RishabhJain2018 
img_20181108_004200

### PR DESCRIPTION
**Pull request by Sanji515**
### **Fix #1893**
Now we can edit host team name in "Host Teams" like we were doing in "Participate Team"

### CODE REVIEW

**Comment by Sanji515** [comment]
@RishabhJain2018 I've fixed the errors. Please review

**Comment by Sanji515** [comment]
@RishabhJain2018 @deshraj Please review.

File: apps/hosts/migrations/0004_auto_20190111_0007.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# -*- coding: utf-8 -*-
```
Comment (RishabhJain2018) [inline apps/hosts/migrations/0004_auto_20190111_0007.py]: Can you please name the migrations as `Add_blank_in_host_team_URL.py`
You can name the migrations while generating them using
`python manage.py makemigrations --name=Add_blank_in_host_team_URL --settings=settings.dev`

File: frontend/src/js/controllers/challengeHostTeamsCtrl.js [hunk 2]
```diff
@@ -37,7 +37,6 @@
         // stop loader
         vm.stopLoader = loaderService.stopLoader;
 
-
```
Comment (RishabhJain2018) [inline frontend/src/js/controllers/challengeHostTeamsCtrl.js]: Please remove this change!

File: frontend/src/js/controllers/challengeHostTeamsCtrl.js [hunk 3]
```diff
@@ -138,6 +137,86 @@
...
 
+
+        vm.showMdDialog = function(ev, hostTeamId) {
+            vm.hostTeamId = hostTeamId;
+            parameters.url = 'hosts/challenge_host_team/' + hostTeamId;
+            parameters.method = 'GET';
+            parameters.token = userKey;
+            parameters.callback = {
+                onSuccess: function(response) {
+                    vm.team.name_modal = response.data.team_name;
```
Comment (RishabhJain2018) [inline frontend/src/js/controllers/challengeHostTeamsCtrl.js]: Please use CamelCase.
`vm.team.name_modal` -> `vm.team.TeamName`

File: frontend/src/js/controllers/challengeHostTeamsCtrl.js [hunk 4]
```diff
@@ -138,6 +137,86 @@
...
+                    vm.team.url_modal = response.data.team_url;
```
Comment (RishabhJain2018) [inline frontend/src/js/controllers/challengeHostTeamsCtrl.js]: `vm.team.url_modal -> vm.team.TeamURL`

File: frontend/src/js/controllers/challengeHostTeamsCtrl.js [hunk 5]
```diff
@@ -138,6 +137,86 @@
...
+            $mdDialog.show({
+                scope: $scope,
+                preserveScope: true,
+                targetEvent: ev,
+                templateUrl: 'dist/views/web/edit-host-teams.html'
+            });
+        };
+
+
+        vm.updateHostTeamData = function(updateHostTeamDataForm) {
```
Comment (RishabhJain2018) [inline frontend/src/js/controllers/challengeHostTeamsCtrl.js]: `vm.updateHostTeamData` -> `vm.updateChallengeHostTeamData`

`updateHostTeamDataForm` -> `updateChallengeHostTeamDataForm`

File: frontend/src/js/controllers/challengeHostTeamsCtrl.js [hunk 6]
```diff
@@ -138,6 +137,86 @@
...
+            if (updateHostTeamDataForm) {
```
Comment (RishabhJain2018) [inline frontend/src/js/controllers/challengeHostTeamsCtrl.js]: `updateHostTeamDataForm` -> `updateChallengeHostTeamDataForm`

File: frontend/src/js/controllers/challengeHostTeamsCtrl.js [hunk 7]
```diff
@@ -138,6 +137,86 @@
...
+            var parameters = {};
+            parameters.url = 'hosts/challenge_host_team/' + vm.hostTeamId;
+            parameters.method = 'PATCH';
+            parameters.data = {
+                "team_name": vm.team.name_modal,
+                "team_url": vm.team.url_modal
```
Comment (RishabhJain2018) [inline frontend/src/js/controllers/challengeHostTeamsCtrl.js]: Please modify it as per above variable names.

**Comment by RishabhJain2018** [comment]
@Sanji515 Please remove [WIP] from the title once you are done with the changes.

File: apps/hosts/migrations/0004_auto_20190111_0007.py (hunk 1 above)
Comment (Sanji515) [reply apps/hosts/migrations/0004_auto_20190111_0007.py]: @RishabhJain2018 done.

File: frontend/src/js/controllers/challengeHostTeamsCtrl.js (hunk 2 above)
Comment (Sanji515) [reply frontend/src/js/controllers/challengeHostTeamsCtrl.js]: done.

File: frontend/src/js/controllers/challengeHostTeamsCtrl.js (hunk 3 above)
Comment (Sanji515) [reply frontend/src/js/controllers/challengeHostTeamsCtrl.js]: done.

File: frontend/src/js/controllers/challengeHostTeamsCtrl.js (hunk 4 above)
Comment (Sanji515) [reply frontend/src/js/controllers/challengeHostTeamsCtrl.js]: done.

File: frontend/src/js/controllers/challengeHostTeamsCtrl.js (hunk 5 above)
Comment (Sanji515) [reply frontend/src/js/controllers/challengeHostTeamsCtrl.js]: done.

File: frontend/src/js/controllers/challengeHostTeamsCtrl.js (hunk 6 above)
Comment (Sanji515) [reply frontend/src/js/controllers/challengeHostTeamsCtrl.js]: done.

File: frontend/src/js/controllers/challengeHostTeamsCtrl.js (hunk 7 above)
Comment (Sanji515) [reply frontend/src/js/controllers/challengeHostTeamsCtrl.js]: done.

**Comment by Sanji515** [comment]
@RishabhJain2018 I've done all the changes as you have mentioned.

File: frontend/src/js/controllers/challengeHostTeamsCtrl.js [hunk 8]
```diff
@@ -34,10 +34,6 @@
          // show loader
         vm.startLoader = loaderService.startLoader;
 
-        // stop loader
-        vm.stopLoader = loaderService.stopLoader;
```
Comment (RishabhJain2018) [inline frontend/src/js/controllers/challengeHostTeamsCtrl.js]: Please don't remove this line.

Comment (Sanji515) [reply frontend/src/js/controllers/challengeHostTeamsCtrl.js]: @RishabhJain2018 done.

**Comment by Sanji515** [comment]
@RishabhJain2018 Done with the changes, Please review.

**Pull request merged**